src/state/helpers.ts

```diff
@@ -38,9 +48,12 @@ export const getEnabledSwappers = (
       ArbitrumBridge &&
       (!isCrossAccountTrade || isCrossAccountTradeSupported(SwapperName.ArbitrumBridge)),
     [SwapperName.Portals]:
-      Portals && (!isCrossAccountTrade || isCrossAccountTradeSupported(SwapperName.Portals)),
+      PortalsSwap && (!isCrossAccountTrade || isCrossAccountTradeSupported(SwapperName.Portals)),
     [SwapperName.Chainflip]:
-      Chainflip && (!isCrossAccountTrade || isCrossAccountTradeSupported(SwapperName.Chainflip)),
+      ChainflipSwap &&
+      (!isCrossAccountTrade || isCrossAccountTradeSupported(SwapperName.Chainflip)),
+    [SwapperName.Jupiter]:
+      JupiterSwap && (!isCrossAccountTrade || isCrossAccountTradeSupported(SwapperName.Jupiter)),
```

Comment:
i.e this should be something like
```suggestion
      JupiterSwap && (!isCrossAccountTrade || (!isSolBuyAssetId && isCrossAccountTradeSupported(SwapperName.Jupiter))),
```

Where `isSolBuyAssetId` is passed to `getEnabledSwappers` - this way, we can remove all the additional `isManualReceiveAddress` plumbing, and most importantly ensure that we stick to current swapper flow i.e no wallet connected is able to yield rates regardless of cross-account support, while connecting a wallet yields only rates for supported cross-account routes

Reply:
Fixed in [0001e53](https://github.com/shapeshift/web/pull/8120/commits/0001e5316ccd4395724ab75e36a101390df9b1d8)

Reply:
Much love very cleaner ser!